helm/mongodb/templates/service.yaml

```diff
@@ -1,12 +1,19 @@
+{{- $values :=  .Values }}
+{{- $chart := .Chart }}
+
+{{- range tuple "0" "1" "2" }}
+---
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{.Values.service.name}} 
+  name: {{ $values.service.name }}-{{ . }}
   labels:
-    chart: "{{ .Chart.Name | trunc 63 }}"
-    chartVersion: "{{ .Chart.Version | trunc 63 }}"
-    app: {{.Values.service.name}} 
+    chart: "{{ $chart.Name | trunc 63 }}"
+    chartVersion: "{{ $chart.Version | trunc 63 }}"
+    app: {{ $values.service.name }}-{{ . }}
     visualize: "true"
+  annotations:
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
```

Comment:
It would be nice to put the explanation from the review on this annotation, to let anyone know if wondering.